### PR TITLE
AddRoles

### PR DESCRIPTION
**Pull request by Dinexpod**

### CODE REVIEW

File: src/main/java/mate/academy/web/filters/UserFilter.java [hunk 1]
```diff
@@ -46,26 +52,55 @@ public void doFilter(ServletRequest servletRequest,
...
+                    || path.equals("/servlet/registration")
+                    || path.equals("/servlet/404")
+                    || path.equals("/servlet/403")) {
                 processAuth(req, resp, filterChain);
             } else {
                 prosessUnauth(req, resp);
             }
         } else {
             User u = userDao.findByToken(token);
+            u.addRole("ADMIN"); ///////////////////////////////////////////////
```
Comment (YatKat) [inline src/main/java/mate/academy/web/filters/UserFilter.java]: May be you shold use RoleRepository here.

Comment (Dinexpod) [reply src/main/java/mate/academy/web/filters/UserFilter.java]: Yes, you right) thanks

File: src/main/java/mate/academy/Main.java [hunk 2]
```diff
@@ -27,6 +30,7 @@ public static void main(String[] args) {
         DeveloperService developerS = new DeveloperServiceImpl(new DeveloperDaoImpl(connection));
         ProjectService projectS = new ProjectServiceImpl(new ProjectDaoImpl(connection));
         SkillService skillS = new SkillServiceImpl(new SkillDaoImpl(connection));
+        RoleService roleS = new RoleServiceImpl(new RoleDaoImpl(connection));
```
Comment (boroda4436) [inline src/main/java/mate/academy/Main.java]: @Dinexpod it's not very good to use S as a Service. Just FYI

File: src/main/java/mate/academy/models/Role.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "Role{" +
+                "id=" + id +
+                ", role='" + role + '\'' +
+                '}';
+    }
+}
```
Comment (boroda4436) [inline src/main/java/mate/academy/models/Role.java]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/403.jsp [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+<html>
+<head>
+    <title>Process denied</title>
+</head>
+<%@include file="header1.jsp"%>
+
+<body>
+<h1>Process denied</h1>
+</body>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/403.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/404.jsp [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    <title>Not found</title>
+</head>
+<%@include file="header1.jsp"%>
+
+<body>
+
+
+<h1>404</h1>
+</body>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/404.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/admin.jsp [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+<html>
+<head>
+    <title>admin</title>
+</head>
+<%@include file="header1.jsp"%>
+
+<body>
+<h1>ADMIN</h1>
+</body>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/admin.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/header1.jsp [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+<a href="${pageContext.request.contextPath}/servlet/home">
+    <input type="submit" value=" home ">
+</a>
+
+<a href="${pageContext.request.contextPath}/servlet/login">
+    <input type="submit" value=" login ">
+</a>
+
+</body>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/header1.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/header2.jsp [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    <title>header2</title>
+</head>
+<%@include file="header1.jsp"%>
+
+<body>
+
+
+<h1>HEADER 2</h1>
+</body>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/header2.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/home.jsp [hunk 9]
```diff
@@ -13,30 +13,34 @@
...
-    </a>
+        </body>
 
-    </body>
+    </form>
 
 </div>
 
-</html>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/home.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 10]
```diff
@@ -13,36 +13,27 @@
...
             <input type="submit" value=" Go to registration ">
         </form>
     </a>
-
     </body>
-
 </div>
-
-</html>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/login.jsp]: empty line at the EOF

File: src/main/webapp/WEB-INF/views/registration.jsp [hunk 11]
```diff
@@ -13,42 +13,31 @@
...
             <input type="submit" value=" Go to login ">
         </form>
     </a>
-
     </body>
-
 </div>
-
-</html>
+</html>
```
Comment (boroda4436) [inline src/main/webapp/WEB-INF/views/registration.jsp]: empty line at the EOF

File: src/main/java/mate/academy/Main.java (hunk 2 above)
Comment (Dinexpod) [reply src/main/java/mate/academy/Main.java]: +

File: src/main/java/mate/academy/models/Role.java (hunk 3 above)
Comment (Dinexpod) [reply src/main/java/mate/academy/models/Role.java]: +

File: src/main/webapp/WEB-INF/views/403.jsp (hunk 4 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/403.jsp]: +

File: src/main/webapp/WEB-INF/views/admin.jsp (hunk 6 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/admin.jsp]: +

File: src/main/webapp/WEB-INF/views/header1.jsp (hunk 7 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/header1.jsp]: +

File: src/main/webapp/WEB-INF/views/header2.jsp (hunk 8 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/header2.jsp]: +

File: src/main/webapp/WEB-INF/views/home.jsp (hunk 9 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/home.jsp]: +

File: src/main/webapp/WEB-INF/views/login.jsp (hunk 10 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/login.jsp]: +

File: src/main/webapp/WEB-INF/views/registration.jsp (hunk 11 above)
Comment (Dinexpod) [reply src/main/webapp/WEB-INF/views/registration.jsp]: +

**Pull request merged**